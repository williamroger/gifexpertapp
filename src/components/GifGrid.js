import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then(imgs => setImages(imgs));
  }, [category]);


  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map(image => (
          <GifGridItem 
            key={image.id}
            {...image}
          />
        ))}
      </div>
    </>
  );
};
