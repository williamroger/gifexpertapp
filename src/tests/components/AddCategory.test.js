import React from 'react';
import '@testing-library/jest-dom';

import { AddCategory } from '../../components/AddCategory';
import { shallow } from 'enzyme';

describe('Tests form component AddCategory', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should be component AddCategory', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be change value of input', () => {
    const input = wrapper.find('input');
    const value = 'Star Wars';

    input.simulate('change', { target: { value }});

    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('should not be send value on submit form', () => {
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should be called setCategories and cleaned input value on submit', () => {
    const input = wrapper.find('input');
    const value = 'Big Bang Theory';

    input.simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  });
});