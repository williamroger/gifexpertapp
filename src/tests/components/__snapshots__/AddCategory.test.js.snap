// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests form component AddCategory should be component AddCategory 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
